drivers/i2c/Kconfig

```diff
@@ -218,7 +220,7 @@ config I2C_1
 config I2C_1_NAME
 	string "Port 1 device name"
 	default "I2C_1"
-	depends on I2C_1
+	depends on I2C_1 && !HAS_DTS_I2C
 
 config I2C_1_DEFAULT_CFG
```

Comment:
I want to suggest to sort out such issues as default configuration now, not later. Introducing new DTS based I2C config is a good chance to get rid of this config option. It is quite ugly. It's also not clear what DTS based I2C drivers should do with it.

As @erwango suggested one way to go would be to add in DTS specific parameters we want to configure. The other approach, mentioned by @ydamigos, would be not to call i2c_configure() in I2C driver initialization function but relay on a user calling i2c_configure() in the application. In this case we would not need to define any additional parameters in DTS. I'm personally in favor of the second approach.

Once this PR is merged I can update Atmel I2C drivers to use DTS.